api/src/test/java/com/epam/pipeline/controller/region/CloudRegionControllerTest.java

```diff
@@ -0,0 +1,460 @@
+    private AwsRegion awsRegion;
+    private AWSRegionDTO awsRegionDTO;
+    private AzureRegion azureRegion;
+    private AzureRegionDTO azureRegionDTO;
+    private GCPRegion gcpRegion;
+    private GCPRegionDTO gcpRegionDTO;
+    private ResponseResult<AwsRegion> expectedAwsRegionResult;
+    private ResponseResult<AzureRegion> expectedAzureResult;
+    private ResponseResult<GCPRegion> expectedGcpResult;
+
+    @Autowired
+    private CloudRegionApiService mockCloudRegionApiService;
+
+    @Before
+    public void setUp() {
+        awsRegion = RegionCreatorUtils.getDefaultAwsRegion();
+        azureRegion = RegionCreatorUtils.getDefaultAzureRegion();
+        gcpRegion = RegionCreatorUtils.getDefaultGcpRegion();
+
+        awsRegionDTO = RegionCreatorUtils.getDefaultAwsRegionDTO();
+        azureRegionDTO = RegionCreatorUtils.getDefaultAzureRegionDTO();
+        gcpRegionDTO = RegionCreatorUtils.getDefaultGcpRegionDTO();
+
+        expectedAwsRegionResult = ControllerTestUtils.buildExpectedResult(awsRegion);
+        expectedAzureResult = ControllerTestUtils.buildExpectedResult(azureRegion);
+        expectedGcpResult = ControllerTestUtils.buildExpectedResult(gcpRegion);
+    }
+
+    @Test
+    public void shouldFailLoadProvidersForUnauthorizedUser() throws Exception {
+        mvc().perform(get(LOAD_PROVIDERS_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadProviders() throws Exception {
+        final List<CloudProvider> cloudProviders = Collections.singletonList(CloudProvider.AWS);
+
+        Mockito.doReturn(cloudProviders).when(mockCloudRegionApiService).loadProviders();
+
+        final MvcResult mvcResult = mvc().perform(get(LOAD_PROVIDERS_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockCloudRegionApiService).loadProviders();
+
+        final ResponseResult<List<CloudProvider>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudProviders);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
+                new TypeReference<Result<List<CloudProvider>>>() { });
+    }
+
+    @Test
+    public void shouldFailLoadAllForUnauthorizedUser() throws Exception {
+        mvc().perform(get(REGION_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAll() throws Exception {
+        final List<AwsRegion> cloudRegions = Collections.singletonList(awsRegion);
+
+        Mockito.doReturn(cloudRegions).when(mockCloudRegionApiService).loadAll();
+
+        final MvcResult mvcResult = mvc().perform(get(REGION_URL)
```

Comment:
Maybe we should consider implementation common method to avoid such routine? Also, it would be grate to add same method for unauthorized case.